### PR TITLE
Apache Airflow update: Removed RDS for now, added doc and refactored codes

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:* #651

*Description of changes:*
- Refactoring code to include full custom helm values for S3, ALB and EFS implementations
- Commented out RDS implementation (as we await for RDS Provider PR to merge)
- Added documentations.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/apache-airflow/index.ts [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    constructor(props?: AirflowAddOnProps) {
+        super({...defaultProps, ...props});
+        this.options = this.props as AirflowAddOnProps;
+    }
+
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+
+        // Create Namespace
+        const ns = createNamespace(this.options.namespace!, cluster, true, true);
```
Comment (shapirov103) [inline lib/addons/apache-airflow/index.ts:105]: is there a case when they want to install into an existing namespace? we can call out in the documentation. Generally, we also add a `createNamespace` attribute

File: lib/addons/apache-airflow/index.ts [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+        return Promise.resolve(chart);
+    }
+}
+
+/**
+ * populateValues populates the appropriate values used to customize the Helm chart
+ * @param helmOptions User provided values to customize the chart
+ */
+function populateValues(clusterInfo: ClusterInfo, ns: KubernetesManifest, helmOptions: AirflowAddOnProps): Values {
```
Comment (shapirov103) [inline lib/addons/apache-airflow/index.ts:119]: Our general approach on reconciling the values passed to the helm chart by the user explicitly and values that we set ourselves is to only set the values for keys that are not supplied by the customers - customer explicit settings takes precedence.  it is generally achieved by merging onto the customer values like [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/appmesh/index.ts#L101)

File: lib/addons/apache-airflow/index.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        name: `${sa.serviceAccountName}`
+    });
+    
+    return values;
+}
+
+/**
+ * TODO: Helper function to set up the Metadata DB with RDS
+*/
+// function setUpDatabase(values: Values, dbConfig: {username: string; password: string; dbName: string;} | undefined): Values {
```
Comment (shapirov103) [inline lib/addons/apache-airflow/index.ts:282]: Drop the comments

**Review by shapirov103 — COMMENTED** [review]
@youngjeong46 looks great, a few comments for you. 

Testing - what are we doing for it? Do you want to add it to the e2e test?

File: lib/addons/apache-airflow/index.ts (hunk 3 above)
Comment (youngjeong46) [reply lib/addons/apache-airflow/index.ts]: addressed

File: lib/addons/apache-airflow/index.ts [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    // dbConfig?: {
+    //     username: string,
+    //     password: string,
+    //     dbName: string 
+    // }
+    
+    /**
+     * Enable Logging with S3
+     */
+    enableLogging?: boolean,
```
[the remark this answers is not in the document]
Comment (youngjeong46) [reply lib/addons/apache-airflow/index.ts]: addressed.

File: lib/addons/apache-airflow/index.ts [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    enableEfs?: boolean,
+
+    /**
+     * Names of the EFS File System provider named resources (@see CreateEfsFileSystemProvider, @see LookupEfsFileSystemProvider).
+     * EFS File System provider is registered as named resource providers with the EksBlueprintProps.
+     * This is required if EFS is enabled
+     */
+    efsFileSystem?: string,
+
+    /**
```
[the remark this answers is not in the document]
Comment (youngjeong46) [reply lib/addons/apache-airflow/index.ts]: addressed.

**Comment by youngjeong46** [comment]
> @youngjeong46 looks great, a few comments for you.
> 
> Testing - what are we doing for it? Do you want to add it to the e2e test?

I have added e2e. Still testing a few other stuff, so will address rest of the comments.

**Comment by shapirov103** [comment]
> I have added e2e. Still testing a few other stuff, so will address rest of the comments.
Thank you @youngjeong46 , I will release tonight after e2e.

File: lib/addons/apache-airflow/index.ts (hunk 2 above)
Comment (youngjeong46) [reply lib/addons/apache-airflow/index.ts:115]: This may need to be addressed in the next round of PR - there's a bit of refactoring needed here.

Comment (shapirov103) [reply lib/addons/apache-airflow/index.ts:115]: Sounds good. I am going to release now, please continue on this.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**